apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -27,13 +27,16 @@
     var f = new navigator.mozL10n.DateTimeFormat();
     var _ = navigator.mozL10n.get;
 
+    /* We use the span tag only as a marker to remove the am/pm component of
+     * the string as we don't show it in the lockscreen. */
     var timeFormat = this._timeFormat.replace('%p', '<span>%p</span>');
     var dateFormat = _('longDateFormat');
 
-    var timeHTML = f.localeFormat(now, timeFormat);
+    var timeText =
+      f.localeFormat(now, timeFormat).replace(/\s<span>.*<\/span>/, '');
```

Comment:
Thanks for elaboration. I think I know what you're saying, but I just worry about if anyone could get this 3 months later after this patch got landed. Maybe we could modify the comment at line 30 to add that your motivation is because we don't show AM/PM anymore but API can do nothing for that, so it's a workaround. And mark it as `XXX` and ref to this bug number.